File: test1-main (10)/test1-main/Integrations/admin/getAccessToken.js
import axios from "axios";


async function getAccessToken(loginId = 0, clientCredentials = 'MTdiZjc4NzYtNThhNy00YzBkLTgzMmEtNjAzNmMxNWEwNWE3Olk1SFdFSVMxZUdQUjdMb3YyUS9MbGNiZ3NPc3Z5anhS') {
let data = JSON.stringify({
  "grant_type": "client_credentials",
  "user_login_id": loginId
});

let config = {
  method: 'post',
  maxBodyLength: Infinity,
  url: 'https://secure-wms.com/AuthServer/api/Token',
  headers: { 
    'Host': 'secure-wms.com', 
    'Connection': 'keep-alive', 
    'Content-Type': 'application/json', 
    'Accept': 'application/json', 
    'Authorization': `Basic ${clientCredentials}`, 
    'Accept-Encoding': 'gzip,deflate,sdch', 
    'Accept-Language': 'en-US,en;q=0.8'
  },
  data : data
};

const response = await axios.request(config)

return response.data

}

export default getAccessToken;

File: test1-main/deprecated/get_access_token.py
import http.client
import json


def get_access_token():
    conn = http.client.HTTPSConnection("box.secure-wms.com")
    payload = json.dumps({
    "grant_type": "client_credentials",
    "user_login_id": "246" # we need a different one
    })
    headers = {
    'Host': 'box.secure-wms.com',
    'Connection': 'keep-alive',
    'Content-Type': 'application/json',
    'Accept': 'application/json',
    'Authorization': 'Basic YjFjZGZhNTktOGRkMS00ZDU1LTkyMjAtYTQxNWZlMWQ3OTIyOnNvRElmTjdXbENyQTJmVzdHYnNrcWRGZDNhWlVpcXQ3', #this is urlencoded clientid:client secret
    'Accept-Encoding': 'gzip,deflate,sdch',
    'Accept-Language': 'en-US,en;q=0.8'
    }
    conn.request("POST", "/AuthServer/api/Token", payload, headers)
    res = conn.getresponse()
    data = res.read()
    print(data.decode("utf-8"))

    return json.loads(data)


if __name__ == "__main__":
    get_access_token()
